src/codegen/doc/format.rs

```diff
@@ -135,18 +135,34 @@ fn replace_symbols(
     env: &Env,
     in_type: Option<(&TypeId, Option<LocationInObject>)>,
 ) -> String {
-    // We run gi_docgen first because it's super picky about the types it replaces
-    let out = gi_docgen::replace_c_types(input, env, in_type);
-    let out = replace_c_types(&out, env, in_type);
-    // this has to be done after gi_docgen replaced the various types it knows as it uses `@` in it's linking format
-    PARAM_SYMBOL
-        .replace_all(&out, |caps: &Captures<'_>| format!("`{}`", &caps[2]))
-        .to_string()
+    if env.config.use_gi_docgen {
+        let out = gi_docgen::replace_c_types(input, env, in_type);
+        let out = GI_DOCGEN_SYMBOL.replace_all(&out, |caps: &Captures<'_>| match &caps[2] {
+            "TRUE" => "[`true`]".to_string(),
+            "FALSE" => "[`false`]".to_string(),
+            "NULL" => "[`None`]".to_string(),
+            symbol_name => match &caps[1] {
+                // Opt-in only for the %SYMBOLS, @/# causes breakages
```

Comment:
Can we factor this out?

Reply:
we can't because you use a different regex, but we can merge some parts of it somehow

Reply:
you here as in the gtk-doc parser